article/app/AppLoader.scala

```diff
@@ -20,14 +20,17 @@ import play.api.routing.Router
 import play.api._
 import services.{NewspaperBooksAndSectionsAutoRefresh, OphanApi}
 import router.Routes
+import conf.Configuration
 
 class AppLoader extends FrontendApplicationLoader {
   override def buildComponents(context: Context): FrontendComponents = new BuiltInComponentsFromContext(context) with AppComponents
 }
 
 trait AppComponents extends FrontendComponents with ArticleControllers {
 
-  lazy val capiHttpClient: HttpClient = wire[CapiHttpClient]
+  lazy val _isDevInfra = Configuration.environment.stage.equalsIgnoreCase("DEVINFRA")
+  lazy val capiHttpClient: HttpClient = if (_isDevInfra) wire[MockHttpClient] else wire[CapiHttpClient]
```

Comment:
I don't really like conditional in the production code. Can you explain why and how you want to use article with mock capi?